apps/docs/docs/nest.md

```diff
@@ -28,6 +29,42 @@ The `@Api` decorator takes the route, defines the path and method for the contro
 
 It also injects "appRoute" into the req object, allowing the `@ApiDecorator` decorator automatically parse and check the query and body parameters.
 
+
+## Response Return Type Safety
+
+You have two options to ensure HTTP type safety on your Nest Controllers: 
+
+- `ControllerShape` as shown above: 
+  - Your controller can implement the `ControllerShape` derived from `typeof s.controllerShape`
+  - This ensures your controller methods also align with the base interface shapes
+- `ResponseShapes`:
+  - Your controller can utilize `typeof s.responseShapes` in the return type. For example:
+    ```typescript
+    const s = initNestServer(apiBlog);
+    type ControllerShape = typeof s.controllerShape;
+    type RouteShape = typeof s.routeShapes;
+    type ResponseShapes = typeof s.responseShapes; // <- http Responses defined in contract
+
+    @Controller()
+    export class PostController {
+      constructor(private readonly postService: PostService) {}
+
+      @Api(s.route.getPost)
+      async getPost(@ApiDecorator() { params: { id } }: RouteShape['getPost']):
+      Promise<ResponseShapes['getPost']> { // <- return type defined here
+        const post = await this.postService.getPost(id);
+
+        if (!post) {
+          return { status: 404 as const, body: null };
+        }
+
+        return { status: 200 as const, body: post };
```

Comment:
is the as const still needed with the return type? :(

Reply:
You're right! Even better 🥳 
I updated the example